internal/driver/webui.go

```diff
@@ -179,11 +179,24 @@ func defaultWebServer(args *plugin.HTTPServerArgs) error {
 	// https://github.com/google/pprof/pull/348
 	mux := http.NewServeMux()
 	mux.Handle("/ui/", http.StripPrefix("/ui", handler))
-	mux.Handle("/", http.RedirectHandler("/ui/", http.StatusTemporaryRedirect))
+	mux.Handle("/", redirectWithQuery("/ui"))
 	s := &http.Server{Handler: mux}
 	return s.Serve(ln)
 }
 
+func redirectWithQuery(url string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var (
+			urlWithQuery = url
+			queryStr     = r.URL.Query().Encode()
```

Comment:
I feel like using `urlWithQuery := url` instead of var() would fit more in with the style of the code we have so far.

Reply:
I would probably be tempted to write lines 189-195  as:
```
if queryStr := r.URL.Query().Encode(); queryString != "" {
     url = url + "?" + queryStr
}
```

Then use url instead of urlWithQuery (I think that's equivalent).

Reply:
sure, done